src/common/src/types/mod.rs

```diff
@@ -744,9 +744,26 @@ impl ScalarRefImpl<'_> {
             &Self::List(ListRef::ValueRef { val }) => {
                 ser.serialize_struct_or_list(val.to_protobuf_owned())?
             }
-            _ => {
-                panic!("Type is unable to be serialized.")
-            }
+            // For StructRef::Indexed
+            &Self::Struct(val) => ser.serialize_struct_or_list(
```

Comment:
We can impl `to_protobuf_owned` in `StructRef` and reduce the duplicate, what do you think?

```rs
impl StructRef {
      pub fn to_protobuf_owned(&self) -> Vec<u8> {
        let value = ProstStructValue {
            fields: self
                .fields_ref()
                .iter()
                .map(|f| match f {
                    None => {
                        vec![]
                    }
                    Some(s) => s.to_protobuf(),
                })
                .collect_vec(),
        };
        value.encode_to_vec()
    }
}
```

Reply:
> We can impl `to_protobuf_owned` in `StructRef` and reduce the duplicate, what do you think?
> 
> ```rust
> impl StructRef {
>       pub fn to_protobuf_owned(&self) -> Vec<u8> {
>         let value = ProstStructValue {
>             fields: self
>                 .fields_ref()
>                 .iter()
>                 .map(|f| match f {
>                     None => {
>                         vec![]
>                     }
>                     Some(s) => s.to_protobuf(),
>                 })
>                 .collect_vec(),
>         };
>         value.encode_to_vec()
>     }
> }
> ```

I think this is much better 😇.

Reply:
I will rewrite this part.